mango-python/bdgenomics/mango/variantViz.py

```diff
@@ -0,0 +1,192 @@
+#
+# Licensed to Big Data Genomics (BDG) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The BDG licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from collections import Counter, OrderedDict
+from cigar import Cigar
+import matplotlib.pyplot as plt
+import numpy as np
+import matplotlib.ticker as ticker
+import math
+from pyspark.sql import Row
+plt.rcdefaults()
+plt.rcParams['figure.figsize'] = [10, 8]
+
+default_bins=100
+
+class VariantDensityDistribution(object):
+    """ VariantDensityDistribution class.
+    VariantDensityDistribution computes a distribution of positionally binned variant counts.
+    """
+    def __init__(self,ss,variantRDD, start, end, contigName):
+        """
+        Initializes a VariantDensityDistribution class.
+        Computes a distribution of positionally binned variant counts.
+
+        Args:
+            :param ss: SparkContext
+            :param variantRDD: bdgenomics variantRDD
+            :param start: genomic start position
+            :param end:  genomic stop position
+            :param contigName: contig name
+        """
+
+        binSize = 1000000
+        self.data = variantRDD.toDF().filter(
+            (variantRDD.toDF().start > start) &
+            (variantRDD.toDF().start < end) &
+            (variantRDD.toDF().contigName == contigName) ).rdd.map(
+              lambda w: w.asDict()).map(
+               lambda x: (x['contigName'], int(math.floor(float(x['start'])/binSize)))).countByValue()
+
+    def plot(self):
+        """
+        Returns positionally binned counts of variants suitable for plotting
+
+        Returns:
+          A tuple of two lists, the first a list of the start position for each genomic bin
+          and the second a list of corresponding variant counts for each bin.
+        """
+        data = [ (z[0][1],z[1]) for z in self.data.items() ]
```

Comment:
This is confusing. The method is called plot but it doesn't return a plot, just a vector

Reply:
I was trying to be consistent in calling a "plot" function for all of these.
In the example plot function you sent me- where some plot subcomponents like ax are created in the function and then then the plot is actually run in the notebook, a Counter() is returned and also an axis with some formatting.   I didn't see a need for this function to return any specific matplotlib components directly, just data in this case, but just wanted to be consistent.